pages/authzed/concepts/authzed-materialize.mdx

```diff
@@ -1,45 +1,40 @@
 </Callout>
 
-[Dedicated]: ../guides/picking-a-product#dedicated
-
-## What Is AuthZed Materialize?
+AuthZed Materialize is inspired by the Leopard index component described in the [Zanzibar paper](https://zanzibar.tech/2IoYDUFMAE:0:T).
+Much like the concept of a materialized view in relational databases, AuthZed Materialize is a service that computes how permissions change after relationships
+are written, when those relationships affect a subject's membership in a permission set or a set’s permission on a specific resource.
+These permissions are then streamed to your client, so that you can store them as a denormalized table, and then do operations like searching, sorting, and filtering much more efficiently.
 
-Materialize streams computed permission updates to a client.
-Updates occur after a relationship is written that affects a subject's membership in a permission set or a set’s permission on a specific resource.
-The intent is for users to process these updates and store them to form a precomputed and denormalized view of SpiceDB permissions.
+AuthZed Materialize allows you to:
+- speed up `CheckPermission` and `CheckBulkPermissions`
+- speed up `LookupResources` and `LookupSubjects`, especially when there is a large number of resources
+- build authorization-aware UIs, e.g. by providing a filtered and/or sorted list of more than several thousand authorized objects
+- perform ACL filtering in other secondary indexes, like a search index (e.g. ElasticSearch)
 
-## When To Use AuthZed Materialize?
-
-If you need to provide a filtered and/or sorted list of more than several thousand authorized objects or if you need an authorization-aware search index, you probably need Materialize.
-
-The primary use case for Materialize is to denormalize computed permissions into systems that excel at data retrieval operations like searching, sorting, and filtering.
+[Dedicated]: ../guides/picking-a-product#dedicated
 
-There are some authorized object listing scenarios where [LookupResources] or [LookupSubjects], without Materialize, can return a response without a large computational cost.
-Those scenarios include:
+## Limitations
 
-1. Paginating through a list of authorized objects without sorting or filtering ([LookupResources] supports cursor-based pagination, but the list of objects is returned in a non-deterministic order)  
-2. Listing a small set (in the realm of thousands) of ordered or filtered objects.
+If there are:
 
-If you do make a [LookupResources] or [LookupSubjects] request with a significant computational cost, you can expect the request to be slow and to use a large number of system resources, leading to slower response times for other queries.
+- [Caveats]
+- [Wildcard] subject types
+- [.all]
 
-## Current Limitations
+on the path of permissions computed by Materialize, it will error out. However, your schema can still include them.
```

Comment:
To be confirmed: does it error out? It is just my assumption

Reply:
correct, so long it's not in the reachability path of a watched permission, Materialize will not complain. If it does, the hydrator will fail

Reply:
can I mention "hydrator" here or is that too much information?

Reply:
Let's skip if for simplicity